app/queries/admin/users_query.rb

```diff
@@ -4,17 +4,62 @@ class UsersQuery
                    "users.email ILIKE :search OR " \
                    "users.username ILIKE :search".freeze
 
-    def self.call(relation: User.registered, options: {})
-      role, search = options.values_at(:role, :search)
+    # @api public
+    # @param relation [ActiveRecord::Relation<User>]
+    # @param role [String, nil]
+    # @param search [String, nil]
+    # @param roles [Array<String>, nil]
+    def self.call(relation: User.registered, role: nil, search: nil, roles: [])
+      # We are at an interstitial moment where we are exposing both the role and roles param.  We
+      # need to favor one or the other.
+      if role.presence
+        relation = relation.with_role(role, :any)
+      elsif roles.presence
+        relation = filter_roles(relation: relation, roles: roles)
+      end
 
-      relation = relation.with_role(role, :any) if role.presence
       relation = search_relation(relation, search) if search.presence
-
       relation.order(created_at: :desc)
```

Comment:
If users are showing up repeatedly, we might change this to:

```suggestion
      relation.distinct.order(created_at: :desc)
```

Reply:
💥  that's the one

Thank you, I'll make that change in my upcoming PR 🙂